app/services/email_settings_validator.rb

```diff
@@ -0,0 +1,177 @@
+# frozen_string_literal: true
+
+require 'net/imap'
+require 'net/smtp'
+require 'net/pop'
+
+# Usage:
+#
+# begin
+#   EmailSettingsValidator.validate_imap(host: "imap.test.com", port: 999, username: "test@test.com", password: "password")
+#
+#   # or for specific host preset
+#   EmailSettingsValidator.validate_imap(**{ username: "test@gmail.com", password: "test" }.merge(Email.gmail_imap_settings))
+#
+# rescue => err
+#   EmailSettingsValidator.friendly_exception_message(err)
+# end
+class EmailSettingsValidator
+  def self.friendly_exception_message(exception)
```

Comment:
also if we have FRIENDLY_EXCEPTIONS constant we would be able to add a guard to raise an error when an exception is not expected:

```
raise if !FRIENDLY_EXCEPTIONS.include?(exception.class)
```

Reply:
I just want to have the blank rescue there just in case we get an error we are not expecting, because this friendly message method is meant to be used to return a message back to the user.

Reply:
It might be not friendly anymore, but I understand your point :)